apps/shopifyorders/shopify_orders.star

```diff
@@ -0,0 +1,213 @@
+"""
+IMAGE_TRENDS_CONTAINER = """
+iVBORw0KGgoAAAANSUhEUgAAAEAAAAALCAYAAADP9otxAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAABySURBVHgB7dWxDYAgFATQUxmEUSxJGMnGhjiRtVtgyQgygV8aoiZAz4eXEIqrLrkA0KVpMk7RKlE5RZvUt3HoCq4T0tvJxRuMeSsoGbRSPp5f8C5AzCE8wFh2AdyFhzBZfEADvuX3cSl35vpt5BbQte4BvoQ8HyaDvqgAAAAASUVORK5CYII=
+"""
+IMAGE_RECENT_ORDERS = """
+iVBORw0KGgoAAAANSUhEUgAAABAAAAARAQMAAADuYb6HAAAABlBMVEUAAABL/oXORWByAAAAAXRSTlMAQObYZgAAADxJREFUeAFjYGdgYjrAxPiAkfsqkwkHkyELs8YxJrYGpr8MTHwMTOxMjG+YmK4wMCkyMP7/wOTAwMj8EgD1PgqfQoKiPgAAAABJRU5ErkJggg==
+"""
+IMAGE_RECENT_SALES = """
+iVBORw0KGgoAAAANSUhEUgAAABIAAAATCAYAAACdkl3yAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAADeSURBVHgBvVS7EcIwDBVOBmAEagagonB6SgZwxx0TZBU6BqCkTwoqBqANIzAAOSPlAidE/AF8eXfvknOkZz1ZDsAIMMgGaRmrL/JBU9JyMbG3S/7G4z6LFtRSgJKH1qSYEiIVBoU2A4yDwy6bcTHlSzidLaxM+3rGIGjp02LWcIsq1hLZkTH9miYxBYmQg6eCEguezu/Oiji8FVEy9qNjuVXehiezRkI1Hm8hd4s58j6mxteCr2vHBHeU33Ag6R7GDeSzwaFGS6FBiy5L6017lZYkNCS4/RwG/vwf/YQHqM7BNxQY+lYAAAAASUVORK5CYII=
+"""
+
+APP_ID = "shopify_orders"
+
+def api_fetch(counter_id, request_config):
+    cache_key = "{}/{}/{}".format(counter_id, APP_ID, base64.encode(json.encode(request_config)))
+    cached_value = cache.get(cache_key)
+    if cached_value != None:
+        print("Hit! Displaying cached data.")
+        api_response = json.decode(cached_value)
+        return api_response
+    else:
+        print("Miss! Calling Counter API.")
+        url = "{}/tidbyt/api/{}/{}".format(SHOPIFY_COUNTER_API_HOST, counter_id, APP_ID)
+        rep = http.post(url, body = json.encode({"config": request_config}), headers = {"Content-Type": "application/json"})
+        if rep.status_code != 200:
+            print("Counter API request failed with status {}".format(rep.status_code))
+            return None
+        api_response = rep.json()
+        cache.set(cache_key, json.encode(api_response), ttl_seconds = CACHE_TTL)
+        return api_response
+
+def error_view():
+    return render.Root(
+        child = render.Column(
+            children = [
+                render.Image(src = base64.decode(IMAGE_ALIEN)),
+                render.Padding(
+                    pad = (0, 2, 2, 0),
+                    child = render.Marquee(
+                        width = 64,
+                        child = render.Text(content = "We hit a snag. Please check your app.", color = COLOR_WARNING),
+                    ),
+                ),
+            ],
+        ),
+    )
+
+def main(config):
+    counter_id = config.get("counterId")
```

Comment:
One big callout is that we will execute this app with default or potentially blank config to render a preview in the mobile app. You should consider adding a "demo" view if `counterId` is `None` so that folks browsing the mobile app will find a representative preview.